### ISSUE TITLE
Save project settings.

### ISSUE THREAD

**Issue by tyralla** [hydpy-dev/hydpy#47]
So far, _HydPy_ does not yet provide a way to save the project settings made in module _pub_.  If you need to use the same settings later or, for example, in different scripts, it is currently the best practice to write one Python script that makes these settings, for example:

### my_settings.py
```
from hydpy import HydPy, pub
hp = HydPy('my_project')
pub.timegrids = '2000-01-01', '2001-01-01', '1d'
pub.options.checkseries = False
pub.networkmanager.currentdir = 'my_folder'
```

Then, you can import these settings in each script (or in the Python Interpreter) as follows:

### my_script.py
```
from my_settings import *
...
```

The idea is, to improve the string representation of the _pub_ module (which is already the object of the _HydPy_-class [Pub](https://hydpy-dev.github.io/hydpy/pubtools.html#hydpy.core.pubtools.Pub), derived from _Module_).  As always, the string representation should be executable code as far as possible and be applicable to restore the current configuration of the object.  To achieve this goal, we would also need to improve the string representations of some other classes (e.g. [Options](https://hydpy-dev.github.io/hydpy/optiontools.html#hydpy.core.optiontools.Options) and [ControlManager](https://hydpy-dev.github.io/hydpy/filetools.html#hydpy.core.filetools.ControlManager)).

An additional method of class [Pub](https://hydpy-dev.github.io/hydpy/pubtools.html#hydpy.core.pubtools.Pub) to save the string representation directly into a file would be convenient.

We should discuss if we also want to support saving custom settings.  For example, one could also use the module pub to save some constants relevant for different scripts.  But this would be restricted to using objects already coming with suitable string representations.  So, strings and numbers would be fine, but more complex objects probably not. (We could implement a check into method _setattr_ of module [Pub](https://hydpy-dev.github.io/hydpy/pubtools.html#hydpy.core.pubtools.Pub), which executes the string representation for each given attribute and compares the old with the new object.)

**Comment by BGWKlein** [comment]
The extension of class Options looks great. 

I also thought about adding ignore_timegrids to read_projectsettings but I was not sure about the standard value, False would be my prefered option but it is probably inconsistent with the prefered option write_timegrids = False when writing project settings.

**Comment by tyralla** [comment]
Yes, I am also very uncertain about the "best" default values here (even for both functions).   Is this maybe a hint at not providing any default values so that users are forced to think about the right choice in each situation?

**Comment by BGWKlein** [comment]
I agree with you, providing no default values is probably less convinient for the user but we make sure that the user knows what he is doing.
